bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -108,184 +95,177 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                        || (dataItem.getFilteredCode() != null && dataItem.getFilteredCode() != message.getCode())) {
-                    continue;
-                }
+        for (Thing thing : getThing().getThings()) {
+            BaseDeviceHandler handler = (BaseDeviceHandler) thing.getHandler();
 
-                State state = null;
-
-                switch (dataItem.getDataType()) {
-                    case FLAGS:
-                        state = OnOffType.from(message.getBit(dataItem.getByteType(), dataItem.getBitPos()));
-                        break;
-                    case UINT8:
-                    case UINT16:
-                        state = new DecimalType(message.getUInt(dataItem.getByteType()));
-                        break;
-                    case INT8:
-                    case INT16:
-                        state = new DecimalType(message.getInt(dataItem.getByteType()));
-                        break;
-                    case FLOAT:
-                        float value = message.getFloat();
-                        @Nullable
-                        Unit<?> unit = dataItem.getUnit();
-                        state = (unit == null) ? new DecimalType(value) : new QuantityType<>(value, unit);
-                        break;
-                    case DOWTOD:
-                        break;
-                }
+            if (handler != null) {
+                handler.receiveMessage(message);
+            }
+        }
+    }
 
-                if (state != null) {
-                    logger.debug("Received update for channel '{}': {}", channelId, state);
-                    updateState(channelId, state);
-                }
+    @Override
+    public void connectionStateChanged(ConnectionState state) {
+        if (this.state != state) {
+            this.state = state;
+
+            switch (this.state) {
+                case INITIALIZING:
+                    updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Initializing");
+                    break;
+                case DISCONNECTED:
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Disconnected");
+                    autoReconnect();
+                    break;
+                case CONNECTING:
+                    updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Connecting");
+                    break;
+                case CONNECTED:
+                    updateStatus(ThingStatus.ONLINE);
+                    cancelAutoReconnect();
+                    break;
             }
         }
     }
 
     @Override
     public void handleRemoval() {
-        logger.debug("Removing OpenTherm Gateway handler");
+        logger.debug("Removing OpenThermGateway handler");
         disconnect();
         super.handleRemoval();
     }
 
     @Override
     public void dispose() {
+        logger.debug("Disposing OpenThermGateway handler");
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging
```
void myFun() {
    logger.trace("Enter myfun"); // DONT, DONT, really DONT do that
    doSomething();
    logger.trace("Leave myfun"); // DONT, DONT, really DONT do that
}
```

Reply:
I use logger.debug mainly so I can ask people using the binding to send me their logs to analyze what happened. I personally don't have a ventilation heat recovery unit for example and my boiler only supports certain commands. Also I use a different kind of gateway which makes it hard to figure out what happens without some logging. Being able to trace whats going on, when functions were called, when a connection was dropped or when the background connection thread ended has proven to be very valuable. Afaik this is the main purpose of debug logging?